### PR TITLE
allowing callbacks to be used in active callback

### PR DESCRIPTION
**Pull request by naxvog**
PR for https://github.com/aristath/kirki/issues/961

also made some changes to the way it is determined how and when a loop is left (AND and OR behaviour) avoiding an unnecessary loop-cycle in the process.

### CODE REVIEW

File: includes/class-kirki-field.php [hunk 1]
```diff
@@ -537,8 +537,8 @@ protected function set_tooltip() {
 		 */
 		protected function set_active_callback() {
 
-			if ( is_array( $this->active_callback ) && ! is_callable( $this->active_callback ) ) {
-				if ( isset( $this->active_callback[0] ) && is_array( $this->active_callback[0] ) ) {
+			if ( is_array( $this->active_callback ) ) {
+				if ( isset( $this->active_callback[0] ) ) {
 					$this->required = $this->active_callback;
```
Comment (aristath) [inline includes/class-kirki-field.php:542]: Why were these changed?

If for example now users use something like `'active_callback' => array( 'My_Class', 'my_function' ),` and have a class like this:

``` php
class My_Class {
    public static function my_function( $something ) {
        // code
    }
}
```

it won't properly work...
@naxvog is there any particular reason we can't keep these?

Comment (naxvog) [inline includes/class-kirki-field.php:542]: @aristath You are right, the change at line 540 is unnecessary
The change at line 541 on the other hand should be kept to allow functions in the global scope to be called

Comment (aristath) [inline includes/class-kirki-field.php:542]: > The change at line 541 on the other hand should be kept to allow functions in the global scope to be called

Sounds good 👍

**Comment by aristath** [comment]
merged. 👍
